src/wazuh_db/wdb_global.c

```diff
@@ -1239,3 +1239,318 @@ cJSON* wdb_global_get_agents_by_connection_status(wdb_t* wdb, const char* status
+        }
+        strcat(query, " WHERE ");
+         
+        first_condition_element = true;
+        column = kv_value->column_list;
+        while (NULL != column) {
+            if (column->value.is_pk) {
+                if (first_condition_element) {
+                    strcat(query, column->value.name);
+                    strcat(query, "=?");
+                    first_condition_element = false;
+                } else {
+                    strcat(query, " AND ");
+                    strcat(query, column->value.name);
+                    strcat(query, "=?");
+                }
+            }
+            column = column->next;
+        }
+        strcat(query, ";");
+
+        sqlite3_stmt *stmt = wdb_get_cache_stmt(wdb, query);
+
+        if (NULL != stmt) {
+            int index = 1;
+
+            column = kv_value->column_list;
+            curr = field_values;
+            while (NULL != column) {
+                if (!column->value.is_old_implementation && NULL != *curr) {
+                    if (!column->value.is_pk && strcmp(*curr, "NULL") != 0) {
+                        if (FIELD_INTEGER == column->value.type || FIELD_REAL == column->value.type) {
+                            if (sqlite3_bind_int(stmt, index, atoi(*curr))) {
+                                merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                            }
+                        } else {
+                            if (sqlite3_bind_text(stmt, index, *curr, -1, NULL) != SQLITE_OK) {
+                                merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                            }
+                        }
+                        ++index;
+                    }
+                    ++curr;
+                }
+                column = column->next;
+            }
+
+            column = kv_value->column_list;
+            curr = field_values;
+            while (NULL != column) {
+                if (!column->value.is_old_implementation && NULL != *curr) {
+                    if (column->value.is_pk && strcmp(*curr, "NULL") != 0) {
+                        if (FIELD_INTEGER == column->value.type || FIELD_REAL == column->value.type) {
+                            if (sqlite3_bind_int(stmt, index, atoi(*curr))) {
+                                merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                            }
+                        } else {
+                            if (sqlite3_bind_text(stmt, index, *curr, -1, NULL) != SQLITE_OK) {
+                                merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                            }
+                        }
+                        ++index;
+                    }
+                    ++curr;
+                }
+                column = column->next;
+            }
+            ret_val = SQLITE_DONE == wdb_step(stmt) ? true : false;
```

Comment:
Please, remove the ternary operation and just assign to ret_val: `SQLITE_DONE == wdb_step(stmt)`